backend/utils/expand_bullet_point.py

```
@@ -0,0 +1,189 @@
+        topic_chunks (list): List of text chunks from the same topic
+        topic_heading (str): The heading of the topic for context
+        layer (int): The expansion layer (1=first sub-layer, 2=second sub-layer, etc.)
+
+    Returns:
+        dict: A dictionary with the expanded content
+    """
+    try:
+        # Limit expansion to maximum of 2 layers
+        if layer > 2:
+            return {
+                "error": f"Maximum expansion depth of 2 layers reached. Cannot expand beyond layer 2.",
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)